src/Unleash/Communication/UnleashApiClient.cs

```diff
@@ -183,15 +187,9 @@ public async Task<bool> RegisterClient(ClientRegistration registration, Cancella
         {
             const string requestUri = "client/register";
 
-            var memoryStream = new MemoryStream();
-            jsonSerializer.Serialize(memoryStream, registration);
-            memoryStream.Position = 0;
-
-            const int bufferSize = 1024 * 4;
-
             using (var request = new HttpRequestMessage(HttpMethod.Post, requestUri))
             {
-                request.Content = new StreamContent(memoryStream, bufferSize);
+                request.Content = new StringContent(JsonSerializer.Serialize(registration, options), Encoding.UTF8, "application/json");
```

Comment:
Swap out new streaming. This should have a default buffer of 1024 * 4 bytes anyway